WalletWasabi.Gui/Tabs/SettingsViewModel.cs

```diff
@@ -397,21 +397,146 @@ public ErrorDescriptors ValidateTorSocks5EndPoint()
 		public ErrorDescriptors ValidateBitcoinP2pEndPoint()
 			=> ValidateEndPoint(BitcoinP2pEndPoint, Network.DefaultPort, whiteSpaceOk: true);
 
-		public ErrorDescriptors ValidatePrivacyLevel(string value, bool whiteSpaceOk)
+		public ErrorDescriptors ValidatePrivacyLevel(string privacyLevel, string value, bool whiteSpaceOk)
 		{
-			if (whiteSpaceOk && string.IsNullOrWhiteSpace(value))
+			if (whiteSpaceOk && value is null)
 			{
 				return ErrorDescriptors.Empty;
 			}
 
-			if (uint.TryParse(value, out _))
+			if (uint.TryParse(value, out uint result))
 			{
+				ValidatePrivacyLevelValue(privacyLevel, result, out bool isUnderMinimumPrivacyLevel, out uint minimumPrivacyLevel, out bool isAboveMaximumPrivacyLevel, out uint maximumPrivacyLevel);
+
+				if (isUnderMinimumPrivacyLevel)
+				{
+					return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, $"Minimum privacy for this level is {minimumPrivacyLevel}."));
+				}
+				else if (isAboveMaximumPrivacyLevel)
+				{
+					return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, $"Maximum privacy for this level is {maximumPrivacyLevel}."));
+				}
+
 				return ErrorDescriptors.Empty;
 			}
 
 			return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid privacy level."));
 		}
 
+		private void ValidatePrivacyLevelValue(string privacyLevel, uint result, out bool isUnderMinimumPrivacyLevel, out uint minimumPrivacyLevel, out bool isAboveMaximumPrivacyLevel, out uint maximumPrivacyLevel)
+		{
+			isUnderMinimumPrivacyLevel = false;
+			minimumPrivacyLevel = 2;
+
+			isAboveMaximumPrivacyLevel = false;
+			maximumPrivacyLevel = uint.MaxValue;
+
+			if (privacyLevel == nameof(SomePrivacyLevel))
+			{
+				if (result < 2)
+				{
+					isUnderMinimumPrivacyLevel = true;
+				}
+				else
+				{
+					uint.TryParse(FinePrivacyLevel, out uint fineValue);
+					if (result >= fineValue)
+					{
+						if (fineValue > 2)
+						{
+							isAboveMaximumPrivacyLevel = true;
+							maximumPrivacyLevel = fineValue - 1;
+						}
+						else
+						{
+							uint.TryParse(StrongPrivacyLevel, out uint strongValue);
+							if (result >= strongValue - 1 || result >= maximumPrivacyLevel - 1)
+							{
+								isAboveMaximumPrivacyLevel = true;
+								maximumPrivacyLevel = strongValue > 3 ? strongValue - 2 : maximumPrivacyLevel - 2;
+							}
+						}
+					}
+				}
+			}
+			else if (privacyLevel == nameof(FinePrivacyLevel))
```

Comment:
3 times the same code, extract to a method.

Reply:
Done